scripts/xtensa-build-zephyr.py

```diff
@@ -302,22 +278,60 @@ def check_west_installation():
+	:param west_manifest_path: manifest file to which west is initialized.
+	:type west_manifest_path: pathlib.Path
+	:raises RuntimeError: Raised when west is initialized to wrong manifest file
+	(not SOFs manifest) and script is running in non-interactive mode.
+	"""
+	global west_top
+	message = "West is initialized to manifest other than SOFs!\n"
+	message +=  f"Initialized to manifest: {west_manifest_path}." + "\n"
+	dot_west_directory  = pathlib.Path(west_root_dir.resolve(), ".west")
+	if args.no_interactive:
+		message += f"Try deleting {dot_west_directory } directory and rerun this script."
+		raise RuntimeError(message)
+	question = message + "Reinitialize west to SOF manifest? [Y/n] "
+	print(f"{question}")
+	while True:
+		try:
+			reinitialize_answer = strtobool(input().lower())
+			break
+		except ValueError:
+			sys.stdout.write('Please respond with \'Y\' or \'n\'.\n')
+	if not reinitialize_answer:
+		print("Can not proceed. Reinitialize your west manifest to SOF and rerun this script.")
+		exit(-1)
+	shutil.rmtree(str(dot_west_directory ), ignore_errors=True)
+	execute_command(["west", "init", "-l", f"{SOF_TOP}"], cwd=west_top)
+
 def find_west_workspace():
-	"""[summary] Scans for west workspaces using west topdir command by checking script execution
-	west_top and SOF_TOP directories.
+	"""[summary] Validates whether west workspace had been initialized and points to SOF manifest.
 
-	:return: [description] Returns path when west workspace had been found, otherwise None
-	:rtype: [type] bool, pathlib.Path
+	:return: [description] Returns west topdir and path to west manifest west.yml, otherwise None, None.
+	:rtype: [type] pathlib.Path, pathlib.Path
 	"""
 	global west_top, SOF_TOP
-	paths = [ pathlib.Path(os.getcwd()), SOF_TOP, west_top]
-	for path in paths:
-		if path.is_dir():
-			result = execute_command(["west", "topdir"], capture_output=True,
-						 text=True, check=False, cwd=path)
-		if result.returncode == 0:
-			return pathlib.Path(result.stdout.rstrip(os.linesep))
-	return None
+	west_manifest_path = pathlib.Path(SOF_TOP, "west.yml")
+	result_rootdir = execute_command(["west", "topdir"], capture_output=True, cwd=west_top,
+		timeout=10, check=False)
+	if result_rootdir.returncode != 0:
+		return None, None
+	west_root_dir = pathlib.Path(result_rootdir.stdout.decode().strip('\n')).resolve()
+	result_manifest_dir = execute_command(["west", "config", "manifest.path"], capture_output=True,
+		cwd=west_top, timeout=10, check=True)
+	west_manifest_dir = pathlib.Path(west_root_dir, result_manifest_dir.stdout.decode().strip('\n')).resolve()
+	manifest_file_result = execute_command(["west", "config", "manifest.file"], capture_output=True,
+		cwd=west_top, timeout=10, check=True)
+	returned_manifest_path = pathlib.Path(west_manifest_dir, manifest_file_result.stdout.decode().strip('\n'))
+	if str(returned_manifest_path) != str(west_manifest_path):
+		west_reinitialize(west_root_dir, returned_manifest_path)
```

Comment:
These two lines of code are the only ones that are not acceptable and not compatible with our CI. They must be optional, not run on every build.

Reply:
Marc you are trying to enforce your right in every inventible way possible.
You are trying again to convince me to the very same thing you had written before about "hardcoding manifest path".
Using now far-fetches examples that do not make any sense. Why won't you make example with `cp .git .git2`?
It would be equally senseless. I will not fix this issue because it is not an issue. As I said before I do not agree with you on this and I presented you my arguments. Please stop rolling the same subject over and over again.
This is last time I am answering you to this matter, I am done for. Next steps will be taken with our managers as I am done cooperating with you any further.

Reply:
We had already discussed this, I presented you my arguments which you obviously do not like.
Can you provide concrete problem what is not compatible with our CI?
What is exactly the need to perform
`cp west.yml west2.yml`
`west init -l --mf west2.yml`
operations?

Reply:
Ah I forgot, the log you presented will never happen if you use the script according to instructions.
You forgot to use --no-interactive flag:
`./scripts/xtensa-build-zephyr.py --no-interactive apl`
It will never result in a stucked CI with question "Reinitialize west to SOF manifest? [Y/n] "